funsor/terms.py

```diff
@@ -1249,6 +1274,28 @@ def moment_matching_reduce(op, arg, reduced_vars):
     return instrument.debug_logged(arg.moment_matching_reduce)(op, reduced_vars)
 
 
+class Approximate(Funsor):
+    """
+    Function approximation wrt a semiring.
+    """
+
+    def __init__(self, sum_op, prod_op, model, guide, approx_vars):
```

Comment:
This naming isn't perfect; the `model` here corresponds to Pyro's `model`, but I think the `guide` here corresponds to Pyro's `guide/model`.